app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,50 @@
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def logged_in
+    !current_user.nil?
+  end
+
+  def check_log_in
+    return nil unless current_user.nil?
+
+    flash[:danger] = 'You need to login first.'
+    redirect_to signin_path
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
```

Comment:
- [ ] We should have to methods, one to retrieve the current user and another one to set it. [Requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/current_user_requirement.png).